pyfr/writers/paraview.py

```diff
@@ -22,45 +26,66 @@ class ParaviewWriter(BaseWriter):
-        # element in a partition.
-        for mk, sk in zip(self.mesh_inf, self.soln_inf):
-            off += _write_vtu_header(self.args, self.outf, self.mesh_inf[mk],
-                                     self.soln_inf[sk], off)
-
-        # Write end/start of header/data sections
-        write_s('</UnstructuredGrid>\n<AppendedData encoding="raw">\n_')
+        basename = os.path.splitext(self.outf)[0]
 
-        # Write data "piece"wise
+        parts = defaultdict(list)
         for mk, sk in zip(self.mesh_inf, self.soln_inf):
-            _write_vtu_data(self.args, self.outf, self.cfg, self.mesh[mk],
-                            self.mesh_inf[mk], self.soln[sk],
-                            self.soln_inf[sk])
-
-        # Write .vtu file footer
-        write_s('\n</AppendedData>\n</VTKFile>')
+            part = mk.split('_')[-1]
+
+            if parallel:
+                pfn = '{}_{}.vtu'.format(basename, part)
+            else:
+                pfn = self.outf
+
+            parts[pfn].append((mk, sk))
+
+        for pfn, misil in parts.items():
+            with open(pfn, 'wb') as fh:
+                write_sp = lambda s: fh.write(s.encode('utf-8'))
+                # Write .vtu file header
+                write_sp('<?xml version="1.0" ?>\n<VTKFile '
+                         'byte_order="LittleEndian" '
+                         'type="UnstructuredGrid" '
+                         'version="0.1">\n<UnstructuredGrid>\n')
+                off = 0
+                for mk, sk in misil:
+                    # Write data description header.
+                    # element in a partition.
+                    off += _write_vtu_header(self.args, fh, self.mesh_inf[mk],
+                                             self.soln_inf[sk], off)
+
+                # Write end/start of header/data sections
+                write_sp('</UnstructuredGrid>\n'
+                         '<AppendedData encoding="raw">\n_')
+                for mk, sk in misil:
+                    # Write data "piece"wise
+                    _write_vtu_data(self.args, fh, self.cfg, self.mesh[mk],
+                                    self.mesh_inf[mk], self.soln[sk],
+                                    self.soln_inf[sk])
+
+                # Write .vtu file footer
+                write_sp('\n</AppendedData>\n</VTKFile>')
+        if parallel:
+            with open(self.outf, 'wb') as fh:
+                write_s = lambda s: fh.write(s.encode('utf-8'))
+                write_s('<?xml version="1.0" ?>\n<VTKFile '
+                        'byte_order="LittleEndian" type="PUnstructuredGrid" '
+                        'version="0.1">\n<PUnstructuredGrid>\n')
+                mk, sk = next(iter(parts.values()))[0]
+                _write_vtu_header(self.args, fh, self.mesh_inf[mk],
+                                  self.soln_inf[sk], 0, True)
+                for pfn in parts:
+                    relpath = os.path.relpath(pfn, os.path.dirname(self.outf))
```

Comment:
No, not if you are writing the pvtu files into another directory. In fact, it may always be just '.'